src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -233,32 +227,31 @@ void LoopManager::mark_loop(LinearIR::constExprIt loop_begin_pos,
     }
 }
 
-size_t LoopManager::replace_with_new_loop(const LinearIR& linear_ir,
-                                          LinearIR::constExprIt loop_begin_pos,
-                                          LinearIR::constExprIt loop_end_pos,
-                                          size_t work_amount,
-                                          size_t increment,
-                                          const std::vector<LoopPort>& entries,
-                                          const std::vector<LoopPort>& exits,
-                                          const size_t old_id) {
+size_t LoopManager::replace_with_new_loop(const LinearIR& linear_ir, LinearIR::constExprIt loop_begin_pos, LinearIR::constExprIt loop_end_pos,
+                                          const LoopInfoPtr& loop_info, const size_t old_id) {
     const auto is_bound_explicit_loop_begin = ov::is_type<op::LoopBegin>(loop_begin_pos->get()->get_node());
     const auto is_bound_explicit_loop_end = ov::is_type<op::LoopEnd>(std::prev(loop_end_pos)->get()->get_node());
     OPENVINO_ASSERT((is_bound_explicit_loop_begin && is_bound_explicit_loop_end) || (!is_bound_explicit_loop_begin && !is_bound_explicit_loop_end),
                     "Incorrect LoopBounds!");
     const auto explicit_loop_bounds = is_bound_explicit_loop_begin && is_bound_explicit_loop_end;
+    OPENVINO_ASSERT(std::all_of(m_map.cbegin(), m_map.cend(),
+                    [&loop_info](const std::pair<size_t, LoopInfoPtr>& p) { return loop_info != p.second; }),
+                    "Failed to replace with new Loop: this Loop is already existed!");
```

Comment:
It's not clear why can't I replace a loop with another loop that is already in the map. Seems like we might need smth like that in loop optimizations, e.g. loop fusions can be implemented like that
Also:
```suggestion
                    "Failed to replace with new Loop: this Loop already exists!");
```

Reply:
No-no, I understand why you added this check based on the name and description.
My question is: do you think it can be useful to extend functionality of this method so it can can replace loops with the ones already present in the map. Not necessarily now (I agree to leave it as is), but in principle

Reply:
I think that in general it's incorrect situation: `LoopInfo` contains `LoopPorts`. If we replace loops with the ones already present in the map, we will several loops with the same `LoopInfo` and the same `LoopPorts`. But it's impossible!
(Unfortunately, it's possible at the moment after `InsertSpecificIterations` - but as we discussed offline, probably we will solve this problem soon for brgemm case)